src/btree/bt_curnext.c

```diff
@@ -58,13 +58,12 @@ __cursor_fix_append_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
     } else {
 restart_read:
         WT_RET(__wt_txn_read_upd_list(session, cbt, cbt->ins->upd));
-        if (cbt->upd_value->type == WT_UPDATE_INVALID) {
+        if (cbt->upd_value->type == WT_UPDATE_INVALID ||
+          cbt->upd_value->type == WT_UPDATE_TOMBSTONE) {
             cbt->v = 0;
             cbt->iface.value.data = &cbt->v;
-        } else if (cbt->upd_value->type == WT_UPDATE_TOMBSTONE)
-            cbt->iface.value.data = cbt->upd_value->buf.data;
-        else
-            WT_RET(__wt_value_return(cbt, cbt->upd_value));
+        } else
+            __wt_value_return(cbt, cbt->upd_value);
     }
     cbt->iface.value.size = 1;
```

Comment:
I agree it's untidy. I think here it was a micro-simplification to avoid duplicating the assignment in two of the three branches. I'll fix it here and in bt_curprev.c, which are the only cases I see with grep; if you run across more flag them. (Fixed)